client/my-sites/plugins/plugin-details-CTA/CTA-button.jsx

```diff
@@ -149,6 +150,9 @@ export default function CTAButton( { plugin, hasEligibilityMessages, disabled }
 						return setShowAddCustomDomain( true );
 					}
 					if ( hasEligibilityMessages ) {
+						if ( isEnabled( 'marketplace-plans-page' ) && shouldUpgrade ) {
+							return page( `/plugins/plans/${ selectedSite?.slug }` );
+						}
```

Comment:
I think we need to check the eligibility messages for a specific plan upgrade requirement message. i.e we don't want to show this to users on a plan that supports atomic already.

Reply:
> we don't want to show this to users on a plan that supports atomic already

I think `shouldUpgrade` should cover that scenario. It checks for the install plugins feature and for self hosted Jetpack.

Reply:
That should work. Can you check yaost premium? 

It has an extra pop up asking you to set up a domain before installing. Clicking install bypasses this and goes straight to eligibility window.

![Screenshot 2022-09-15 at 19-15-30 Yoast SEO Premium Plugin ‹ Business - Simple Site — WordPress com](https://user-images.githubusercontent.com/811776/190365611-5e667822-735b-4bef-82f7-d2a05e6ef91f.png)

Reply:
> It has an extra pop up asking you to set up a domain before installing. Clicking install bypasses this and goes straight to eligibility window.

Good find, thanks! I added the same conditional logic to `PluginCustomDomainDialog`'s `onProceed` [here](https://github.com/Automattic/wp-calypso/pull/67770/files#diff-d5ac7cfa2d86b94db92445b2f1e3f311cafbe173848e5b2cc4dd86d550f90ae6R104-R106). Once you click "Install Yoast SEO Premium" it should take you to plugins plans page if you don't have the right plan, show you the "domain warning dialog" if necessary, or take you to checkout if you have everything.